### PR TITLE
Remove observability components

### PR DESCRIPTION
**Pull request by LakshanKarunathilake**
- module-ballerina-observe ( as a standard library )
- module-ballerinai-observe

## Related Issues
- https://github.com/ballerina-platform/ballerina-lang/issues/33195

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-cli/src/main/resources/new_cmd_defaults/manifest-app.toml [hunk 1]
```diff
@@ -2,4 +2,4 @@
 distribution = "DIST_VERSION"
 
 [build-options]
-observabilityIncluded = true
+observabilityIncluded = false
```
Comment (nadundesilva) [inline cli/ballerina-cli/src/main/resources/new_cmd_defaults/manifest-app.toml:5]: This shouldn't be changed as this is the new command's template.

File: cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java [hunk 2]
```diff
@@ -352,7 +352,7 @@ public static void writeBallerinaToml(Path balTomlPath, PackageJson packageJson,
                 + "\"", StandardOpenOption.APPEND);
 
         Files.writeString(balTomlPath, "\n\n[build-options]", StandardOpenOption.APPEND);
-        Files.writeString(balTomlPath, "\nobservabilityIncluded = true\n", StandardOpenOption.APPEND);
+        Files.writeString(balTomlPath, "\nobservabilityIncluded = false\n", StandardOpenOption.APPEND);
```
Comment (nadundesilva) [inline cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CommandUtil.java:355]: This shouldn't change as well as this is also a command output.

**Comment by nadundesilva** [comment]
I think there should be a few more occurances of the config (`observabilityIncluded=true`)

File: gradle/javaLibsProject.gradle [hunk 3]
```diff
@@ -80,9 +80,9 @@ dependencies {
 
     dist project(':ballerina-core')
     dist project(':ballerina-lang')
-    dist project(':ballerina-observability')
+//    dist project(':ballerina-observability')
     dist project(':ballerina-cli')
-    dist project(':ballerina-observability-internal')
+//    dist project(':ballerina-observability-internal')
```
[the remark this answers is not in the document]
Comment (LakshanKarunathilake) [reply gradle/javaLibsProject.gradle]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/34507/commits/6bd61a0e2773eb36a25216ba7e42f6237a44658e

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java [hunk 4]
```diff
@@ -464,7 +464,7 @@ public void testOverrideBuildOptionsOnTomlEdit() {
                 "\n" +
                 "[build-options]\n" +
                 "experimental=true\n" +
-                "observabilityIncluded=true\n" +
+                "observabilityIncluded = false\n" +
```
Comment (nadundesilva) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java:467]: Let's revert this and the assertion below.

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java [hunk 5]
```diff
@@ -438,7 +438,7 @@ public void testOverrideBuildOptions() {
 
         // Verify expected default buildOptions
         Assert.assertFalse(project.buildOptions().skipTests());
-        Assert.assertTrue(project.buildOptions().observabilityIncluded());
+        Assert.assertFalse(project.buildOptions().observabilityIncluded());
```
Comment (nadundesilva) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java:441]: We can revert this by reverting `project-api/project-api-test/src/test/resources/projectWithBuildOptions/Ballerina.toml`

File: project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java [hunk 6]
```diff
@@ -417,7 +417,7 @@ public void testLoadingBuildOptionsFromToml() {
 
         // Verify expected default buildOptions
         Assert.assertTrue(project.buildOptions().skipTests());
-        Assert.assertTrue(project.buildOptions().observabilityIncluded());
+        Assert.assertFalse(project.buildOptions().observabilityIncluded());
```
Comment (nadundesilva) [inline project-api/project-api-test/src/test/java/io/ballerina/projects/test/TestBuildProject.java:420]: Same as other assertion. (can revert this by reverting `project-api/project-api-test/src/test/resources/projectWithBuildOptions/Ballerina.toml`)

**Review by warunalakshitha — CHANGES_REQUESTED** [review]
We have an observability codegen in jballerina backend. So how are we going to test changes to codegen inside ballerina-lang since we move observe tests to the different repo?

Should we use the compiler plugin and move that code to observe stdlib?

**Comment by warunalakshitha** [comment]
Unit test changes are LGTM

**Comment by nadundesilva** [comment]
Is there a way to use a Compiler Plugin to make changes at BIR level ? 🤔 If not, we can't move that logic out of ballerina-lang.

**Comment by hasithaa** [comment]
If we are going to keep the code-gen part, we need to keep test cases as well.

**Comment by nadundesilva** [comment]
Yeah but test cases can't be run since `ballerinai/observe` package is needed to start observability. And `ballerina/observe` is required by `ballerinai/observe`.

If we are to keep the current integration test cases, we need to keep the two modules within ballerina-lang. Or else we need to write new tests only covering the BIR instrumentation.

**Comment by warunalakshitha** [comment]
AFAIK we cannot test observability bir instrumentation since it just run before jvm codegen which will not be part of bir.

**Review by warunalakshitha — APPROVED** [review]
Please add an issue for track whether we can add some tests in lang with dummy module

**Comment by nadundesilva** [comment]
Created an issue to track creating the `ballerinai/observe` dummy module and enabling test cases

**Review by nadundesilva — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]

**Pull request merged**